public/app/plugins/datasource/grafana-pyroscope-datasource/VariableQueryEditor.tsx

```diff
@@ -0,0 +1,162 @@
+
+import { ProfileTypesCascader, useProfileTypes } from './QueryEditor/ProfileTypesCascader';
+import { PhlareDataSource } from './datasource';
+import { Query, VariableQuery } from './types';
+
+export function VariableQueryEditor(props: QueryEditorProps<PhlareDataSource, Query, {}, VariableQuery>) {
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineField
+          label="Query type"
+          labelWidth={20}
+          tooltip={
+            <div>The Prometheus data source plugin provides the following query types for template variables.</div>
+          }
+        >
+          <Select
+            placeholder="Select query type"
+            aria-label="Query type"
+            width={25}
+            options={[
+              { label: 'Profile type', value: 'profileType' as const },
+              { label: 'Label', value: 'label' as const },
+              { label: 'Label value', value: 'labelValue' as const },
+            ]}
+            onChange={(value) => {
+              if (value.value! === 'profileType') {
+                props.onChange({
+                  ...props.query,
+                  type: value.value!,
+                });
+              }
+              if (value.value! === 'label') {
+                props.onChange({
+                  ...props.query,
+                  type: value.value!,
+                  profileTypeId: '',
+                });
+              }
+              if (value.value! === 'labelValue') {
+                props.onChange({
+                  ...props.query,
+                  type: value.value!,
+                  profileTypeId: '',
+                  labelName: '',
+                });
+              }
+            }}
+            value={props.query.type}
+          />
+        </InlineField>
+      </InlineFieldRow>
+
+      {(props.query.type === 'labelValue' || props.query.type === 'label') && (
+        <ProfileTypeRow
+          datasource={props.datasource}
+          initialValue={props.query.profileTypeId}
+          onChange={(val) => {
+            // To make TS happy
+            if (props.query.type === 'label' || props.query.type === 'labelValue') {
+              props.onChange({ ...props.query, profileTypeId: val });
+            }
+          }}
+        />
+      )}
+
+      {props.query.type === 'labelValue' && (
+        <LabelRow
+          value={props.query.labelName}
+          datasource={props.datasource}
+          profileTypeId={props.query.profileTypeId}
+          onChange={(val) => {
+            if (props.query.type === 'labelValue') {
+              props.onChange({ ...props.query, labelName: val });
+            }
+          }}
+          from={props.range?.from.valueOf() || Date.now().valueOf() - 1000 * 60 * 60 * 24}
+          to={props.range?.to.valueOf() || Date.now().valueOf()}
+        />
+      )}
+    </>
+  );
+}
+
+function LabelRow(props: {
+  value: string;
+  datasource: PhlareDataSource;
```

Comment:
Not entirely sure why this is still called `PhlareDataSource`?

Reply:
Will rename it later, I just forgot to rename it.